parmed/openmm/parameters.py

```diff
@@ -468,11 +469,26 @@ def _find_unused_types(self, skip_residues):
 
     @staticmethod
     def _templhasher(residue):
-        if len(residue.atoms) == 1:
-            atom = residue.atoms[0]
-            return hash((atom.atomic_number, atom.type, atom.charge))
-        # TODO implement hash for polyatomic residues
-        return id(residue)
+        """
+        Create a unique hash for each residue and patch template using only properties rendered to OpenMM ffxml.
+        """
+        hash_info = tuple()
+        # Sort tuples of atom properties by atom name
+        if len(residue.atoms) > 0:
+            hash_info += tuple(sorted( [(atom.name, atom.type, str(atom.charge)) for atom in residue.atoms], key=lambda tup: tup[0]))
```

Comment:
This is already the standard sort key, so the `key` keyword seems unnecessary to me.